### PR TITLE
Integrate Android logcat to wptrunner

### PR DESCRIPTION
**Pull request by WeizhongX**
When running test against android targets(clank, webview, weblayer),
pull logcat with command "adb logcat *:D" on the fly, and save
that to result viewer. Logs are only saved for CRASH case for now.

Test: Use geolocation-API/PositionOptions.https.html as an example,
the waterfall run shows the crashlog missing, but it's present in this CL

Bug: 1071136

### CODE REVIEW

**Comment by stephenmcgruer** [comment]
Close/opening to re-run taskcluster now that https://github.com/web-platform-tests/wpt/pull/27355 has landed.

**Comment by stephenmcgruer** [comment]
From tools/ unittests, some style violations to be cleaned up:

```
./wptrunner/wptrunner/browsers/android_webview.py:1:1: F401 'subprocess' imported but unused
./wptrunner/wptrunner/browsers/android_webview.py:3:1: F401 '.base.ExecutorBrowser' imported but unused
./wptrunner/wptrunner/browsers/android_webview.py:6:1: F401 '.chrome_android.AndroidBrowser' imported but unused
./wptrunner/wptrunner/browsers/android_webview.py:7:1: F401 '..webdriver_server.ChromeDriverServer' imported but unused
./wptrunner/wptrunner/browsers/android_webview.py:99:73: W292 no newline at end of file
./wptrunner/wptrunner/browsers/android_weblayer.py:1:1: F401 'subprocess' imported but unused
./wptrunner/wptrunner/browsers/android_weblayer.py:3:1: F401 '.base.Browser' imported but unused
./wptrunner/wptrunner/browsers/android_weblayer.py:3:1: F401 '.base.ExecutorBrowser' imported but unused
./wptrunner/wptrunner/browsers/android_weblayer.py:7:1: F401 '..webdriver_server.ChromeDriverServer' imported but unused
./wptrunner/wptrunner/browsers/android_weblayer.py:101:73: W292 no newline at end of file
./wptrunner/wptrunner/browsers/chrome_android.py:116:9: E722 do not use bare 'except'
```

File: tools/wptrunner/wptrunner/browsers/chrome_android.py
Comment (stephenmcgruer) [inline tools/wptrunner/wptrunner/browsers/chrome_android.py:119]: I'm not sure I love the name `AndroidBrowser`, because there are many browsers on Android (Chrome, Firefox, Edge, Brave, ...) and this code doesn't support anything outside of Chrome. 

What is the purpose of the `ChromeAndroidBrowser` subclass? All it does is override `setup_adb_reverse`, which seems to have the same override in all three subclasses. Couldn't we just move `setup_adb_reverse` to the super-class (`AndroidBrowser`), delete the current `ChromeAndroidBrowser` and rename `AndroidBrowser --> ChromeAndroidBrowser`.

**Review by stephenmcgruer — APPROVED** [review]

**Review by stephenmcgruer — CHANGES_REQUESTED** [review]
(Sorry, did not mean to approve this yet -_-)

File: tools/wptrunner/wptrunner/browsers/chrome_android.py
Comment (stephenmcgruer) [reply tools/wptrunner/wptrunner/browsers/chrome_android.py:119]: Weizhong and I discussed this, and there is a medium bit of refactoring to do to simplify all this further. I don't want to block this PR, so I agreed that we can do the cleanup in a follow-up PR, and for now we will just rename:

`AndroidBrowser --> ChromeAndroidBrowserBase`.

Comment (WeizhongX) [reply tools/wptrunner/wptrunner/browsers/chrome_android.py]: Hi Stephen
Thanks for the review. I have pushed a new commit. Can you pls review again?
Thanks!

File: tools/wptrunner/wptrunner/formatters/chromium.py
Comment (stephenmcgruer) [inline tools/wptrunner/wptrunner/formatters/chromium.py:275]:
```suggestion
        cmd = data.get("command", "")
```

Nit: Don't mix `'` and `"` in the same line. (Preferably not in the same file, but I acknowledge that this file already has mismatching uses!).

**Review by stephenmcgruer — APPROVED** [review]
LGTM modulo one nit and the remaining style problems reported by the CI:

```
./wptrunner/wptrunner/browsers/android_webview.py:5:1: F401 '..webdriver_server.ChromeDriverServer' imported but unused
./wptrunner/wptrunner/browsers/android_webview.py:90:1: W391 blank line at end of file
./wptrunner/wptrunner/browsers/android_weblayer.py:5:1: F401 '..webdriver_server.ChromeDriverServer' imported but unused
./wptrunner/wptrunner/browsers/android_weblayer.py:92:1: W391 blank line at end of file
./wptrunner/wptrunner/browsers/chrome_android.py:206:1: W391 blank line at end of file
```

**Pull request merged**
